packages/core/src/ceramic.ts

```diff
@@ -72,6 +72,7 @@ export interface CeramicConfig {
   cacheDocCommits?: boolean; // adds 'docCacheLimit' additional cache entries if commits can be cached as well
 
   useCentralizedPeerDiscovery?: boolean;
+  restoreDocuments?: boolean;
```

Comment:
I'm a little sad to be adding testing-only configuration options here.  Not the end of the world certainly, but would be nice to avoid if we can do so without causing too much pain in our tests.  I'm actually kind of surprised that this causes issues in the tests - I would have expected the `restoreDocuments()` call to be a no-op because it never actually finds any pinned documents that need restoring.

Reply:
You would be surprised to know, that the whole Inmemory anchor service is just for test purpose only.

Reply:
This is the least intrusive option to avoid re-mocking tests that apparently heavily rely on state store being fully mocked.

Reply:
And, honestly, if the initialisation of Ceramic is done differently, this should go away.

Reply:
> You would be surprised to know, that the whole Inmemory anchor service is for test purpose only.

This is the first InMemoryAnchorService-specific config option to show up in this `CeramicConfig` interface, that's why I'm calling it out.

Reply:
> This is the first InMemoryAnchorService-specific config option to show up in this CeramicConfig interface, that's why I'm calling it out.

I guess there's also `anchorOnCommit`, which doesn't actually show up explicitly here in CeramicConfig, but is implicitly a member as it is passed to the InMemoryAnchorService via this object because `CeramicConfig` allows arbitrary extra options.  I don't really like that either, but maybe doing something similar would be better than polluting this highly user-visible configuration interface with testing-only parameters

Reply:
> This is the least intrusive option to avoid re-mocking tests that apparently heavily rely on state store being fully mocked.

So the issue is that many of the unit tests don't have a real state store, so in order to get the tests to pass without this option, you'd need to update the mocked state store in those tests to expect the initial call to `stateStore.list()` and return an empty list?

Assuming that is indeed the issue, then I think it's fair to want to avoid doing that in a large number of our tests.  I think in that case the thing to do would be to remove this argument from `CeramicConfig` and handle it similarly to how `anchorOnCommit` is handled.  Would be nice to also set it to default to false if we're running in testing mode, similar to how the `useCentralizedPeerDiscovery` default is set

Reply:
> And, honestly, if the initialisation of Ceramic is done differently, this should go away.

How would that work?